oneview_redfish_toolkit/api/computer_system.py

```diff
@@ -236,30 +237,29 @@ def _build_sas_logical_jbods(server_profile_template, storage_blocks):
         return sas_logical_jbods
 
     def _fill_resource_block_members(self,
-                                     server_profile,
                                      drives,
-                                     server_hardware):
+                                     server_hardware,
+                                     server_profile_labels):
         resource_block_uuids = \
-            self._get_resource_block_uuids(server_profile,
-                                           drives,
-                                           server_hardware)
+            self._get_resource_block_uuids(drives,
+                                           server_hardware,
+                                           server_profile_labels)
 
         base_uri = ResourceBlockCollection.BASE_URI + "/{}"
         blocks = self.redfish["Links"]["ResourceBlocks"]
         for resource_block_uuid in resource_block_uuids:
             blocks.append({"@odata.id": base_uri.format(resource_block_uuid)})
 
     def _get_resource_block_uuids(self,
-                                  server_profile,
                                   drives,
-                                  server_hardware):
+                                  server_hardware,
+                                  server_profile_labels):
         resource_block_uuids = list()
         resource_block_uuids.append(server_hardware["uuid"])
 
         # fill with network resource block
-        if server_profile["description"]:
-            spt_id = server_profile["description"].split("/")[-1]
-            resource_block_uuids.append(spt_id)
+        spt_id = server_profile_labels["labels"][0]["name"].replace(" ", "-")
```

Comment:
Once we can have ServerProfiles created directly on OneView without using Redfish Toolkit, I think we need to check if it has just the one expected label on the expected format, to make sure that this label was created by the toolkit and represent and ServerProfileTemplate.

Reply:
And we should put a test covering this case, once is a valid flux.

Reply:
Since the profile is created via toolkit, I think the first label will always be the one we created even if someone uses OneView to create other labels to the server profile later. Please can you just check this behavior? Anyway, I agree that we should confirm if there is a label (someone/script can remove it) and if it represents a serverprofiletemplate (someone/script can change it - I think) , and if not do not add the link.

Reply:
Yes @adriane-cardozo and @victorhugorodrigues  . You're correct. If we created via toolkit, the first label will always be create by using the spt on the post request, independent if there are other labels created on Oneview. But I agree that we need to check if the value passed represents in fact a spt :+1: .